core/src/main/java/it/pagopa/selfcare/mscore/core/EmailService.java

```diff
@@ -47,27 +46,28 @@ public void sendAutocompleteMail(List<String> destinationMail, Map<String, Strin
+    public void sendMailWithContract(File pdf, Institution institution, User user, OnboardingRequest request, String token) {
         List<String> destinationMail;
         Map<String, String> mailParameters;
-        if (InstitutionType.PA == institutionType ||
-                (InstitutionType.GSP == institutionType && request.getProductId().equals("prod-interop") && institution.getOrigin().equals("IPA"))
-                || isApproved) {
-            mailParameters = mailParametersMapper.getOnboardingMailParameter(user, request, token);
-            log.debug(MAIL_PARAMETER_LOG, mailParameters);
-            destinationMail = Objects.nonNull(coreConfig.getDestinationMails()) && !coreConfig.getDestinationMails().isEmpty()
-                    ? coreConfig.getDestinationMails() : List.of(institution.getDigitalAddress());
-            log.info(DESTINATION_MAIL_LOG, destinationMail);
-            emailConnector.sendMail(mailTemplateConfig.getPath(), destinationMail, pdf, request.getProductName(), mailParameters, request.getProductName() + "_accordo_adesione.pdf");
-            log.info("onboarding-contract-email Email successful sent");
-        } else {
-            mailParameters = mailParametersMapper.getOnboardingMailNotificationParameter(user, request, token);
-            log.debug(MAIL_PARAMETER_LOG, mailParameters);
-            destinationMail = mailParametersMapper.getOnboardingNotificationAdminEmail();
-            log.info(DESTINATION_MAIL_LOG, destinationMail);
-            emailConnector.sendMail(mailParametersMapper.getOnboardingNotificationPath(), destinationMail, pdf, request.getProductName(), mailParameters, request.getProductName() + "_accordo_adesione.pdf");
-            log.info("onboarding-complete-email-notification Email successful sent");
-        }
+        mailParameters = mailParametersMapper.getOnboardingMailParameter(user, request, token);
+        log.debug(MAIL_PARAMETER_LOG, mailParameters);
+        destinationMail = Objects.nonNull(coreConfig.getDestinationMails()) && !coreConfig.getDestinationMails().isEmpty()
+                ? coreConfig.getDestinationMails() : List.of(institution.getDigitalAddress());
+        log.info(DESTINATION_MAIL_LOG, destinationMail);
+        emailConnector.sendMail(mailTemplateConfig.getPath(), destinationMail, pdf, request.getProductName(), mailParameters, request.getProductName() + "_accordo_adesione.pdf");
+        log.info("onboarding-contract-email Email successful sent");
+    }
+
+    public void sendMailForApprove(User user, OnboardingRequest request, String token) {
+        List<String> destinationMail;
+        Map<String, String> mailParameters;
+        mailParameters = mailParametersMapper.getOnboardingMailNotificationParameter(user, request, token);
+        log.debug(MAIL_PARAMETER_LOG, mailParameters);
+        destinationMail = mailParametersMapper.getOnboardingNotificationAdminEmail();
```

Comment:
you could create variabile here (List<String> destinationMail = ... ) and avoid to instantiate object null above (List<String> destinationMail;)